src/diffpy/pdfgui/gui/extendedplotframe.py

```diff
@@ -182,9 +182,10 @@ def insertCurve(self, xData, yData, style):
         style -- the way curve should be plotted
         return:  internal reference to the newly added curve
         """
-        stylestr, properties = self.__translateStyles(style)
-        curveRef = self.subplot.plot(xData, yData, stylestr, **properties)[0]
-        self.subplot.legend(**legendBoxProperties())
+        _, properties = self.__translateStyles(style)
+        curveRef = self.subplot.plot(xData, yData, **properties)[0]
```

Comment:
Warning 1: UserWarning: linestyle is redundantly defined by the 'linestyle' keyword argument

`properties` already has `linestyle` shown below:

```
           properties.update({"color": color, "linestyle": lineStyle, "linewidth": lineWidth})
```